### ISSUE TITLE
cover type, channel ?

### ISSUE THREAD

**Issue by uzelac** [ziv1234/hass_dynalite_component#5]
Need bit more explanation on cover subtype, not sure what factor and max/min means.

My covers take two switches, one for up and other for down. Nothing smart.
I just take care that DDRC units don't ask up and down in the same time, since that can happen because Dynalite doesn't understand covers.

At least, that is what i know.

Is there any better way to do covers in Dynalite ?
Because of this limitations, i use timedcover which is also here on github.
Would be much better if Dynalite have something on its own.

Thanks

**Comment by ziv1234** [comment]
Can you show me your config with timedcover?

**Comment by uzelac** [comment]
Sure, there it is as new fail in your component :)
Delete it ASAP, hope you don't mind, couldnt think of better way after i've 'tried (and failed) to format it proprely here in issues.

**Comment by ziv1234** [comment]
Where did you put the file? I cannot see it anywhere on Github

**Comment by uzelac** [comment]
I pressed create new file, and it went... somewhere.
Did this again, hope this link is ok:

https://github.com/uzelac/hass_dynalite_component/blob/687092f99f0e4754496b15d6f417c793c3cb0b6f/timedcover%20example.yaml

**Comment by uzelac** [comment]
As you can see, I have three presets per cover, one for cover to go up, one to go down and third to stop.
That is also how this timedcover works.
Now, with channels, maybe it can be done better, earlier with presets... couldn't find better way.

**Comment by ziv1234** [comment]
Thanks. It works now

**Comment by ziv1234** [comment]
I wrote something today. Mine is the same setup but there is also a channel that can move it up and down. The problem I see with timedcover is that it works if all the action is from HomeAssistant, but I also have wall switches and a Control4 system that may send command.

Hopefully will be able to upload something today or tomorrow

**Comment by uzelac** [comment]
You are right, timedcover wont count time if one instructs cover outside of HA.
It can be done, it can time from channel/preset change to change... Well, it can but it doesn't :). 
Dynalite isnt capable to know position, but HA can.

**Comment by ziv1234** [comment]
I uploaded a new version with an additional template, “timecover”. When you define the templates (or the overrides), you specify:

'timecover':
   channel: 2 # this is optional. I also have a control 4 system that controls the same blind via a channel. If not relevant, use 0
   open: 1 # preset to open
   close: 2 # preset to close
   stop: 4 # preset to stop
   duration: 75 # time to fully open / close
   class: blind # device class. Only affects the visual in HA

I will also add the ability to tilt. My system open tilts when opening the blind briefly and closes them when you close it, so uses the same presets, just for a shorter time.

**Comment by uzelac** [comment]
Great, will give it a try on monday when i return home! :)

**Comment by uzelac** [comment]
Btw maybe even better would be to make:

'timecover':
   channel: 2 # this is optional. I also have a control 4 system that controls the same blind via a channel. If not relevant, use 0
   open: 1 # channel to open
   close: 2 # channel  to close
   duration: 75 # time to fully open / close
   class: blind # device class. Only affects the visual in HA

And so script takes care that up and down are nevert ON at the same time. This way one doesnt need preset for stop.
I believe for me personally it is better with presets, they are already working, but maybe it is more "correct" to use channels

**Comment by ziv1234** [comment]
My mistake: open, close, and stop should be presets, not channels. You need the stop one, since as far as I know, Dynalite doesn’t track the specific preset, but activating a preset causes the action. The channel one is optional as I need it for my setup but it is not mandatory. Set it to 0 to ignore.

The dynalite component already takes care of the fact that they are not up and down at the same time, as an area’s preset is the latest that was selected in that area. I assume that for your blinds, “open”, “close”, and “stop” are all in the same area for a specific blind, right?

**Comment by uzelac** [comment]
Same area, of course.
But I did not mean that.

I meant that component can be made that it needs channels, not presets to work.
In that case:

timecover':
open: 1 # **_channel_** to open
close: 2 # **_channel_** to close
duration: 75 # time to fully open / close
class: blind # device class. Only affects the visual in HA

As i said, maybe it is more correct. 
In that case your script would need to:

Automation OPEN
- close_channel OFF # just in case
- open_channel ON
- time delay
- open_channel OFF

Automation CLOSE
- open_channel OFF # just in case
- close_channel ON
- time delay # count seconds, whatever
- close_channel OFF

**Comment by uzelac** [comment]
What is more important, does your **_timedcover_** script reacts to direct dynalite keypad commands ?
If i use presets on keypad, will it get that and show precise position ?

**Comment by ziv1234** [comment]
my dynalite keypad send the presets so the timecover template monitors these presets and updates the level accordingly. It works in my setup so interested to see whether it works for you.
mine isn't configured with channels this way, as i dont have a direct control (or at least not aware of one) for the open and close channels of the cover itself, so i have to use the presets. 
Let me know if the current one works for your setup. If not, i can try and program it, but since my setup works differently, will be very difficult to test...

**Comment by uzelac** [comment]
Is this OK, to write in .yaml file ? (i use dynalite.yaml with !include in configuration.yaml)

'201':
  name: Guest Room Cover
  template: timecover
  channel: 0
  open: 1
  close: 3
  stop: 2
  duration: 17
  class: blind

**Comment by uzelac** [comment]
Ok, got it, spent some time trying and reading your default configuration.yaml:
And yes, it is a bingo once more :).


Btw, I've put my dynalite.yaml here so someone can learn:
https://github.com/uzelac/hass-dynalite-component-support/blob/master/dynalite.yaml

**Comment by ziv1234** [comment]
thanks

**Comment by ziv1234** [comment]
It should be with the latest version of the library. However, I didn’t update the component
